dartagnan/src/main/java/com/dat3m/dartagnan/encoding/EncodingContext.java

```diff
@@ -111,20 +113,50 @@ public BooleanFormulaManager getBooleanFormulaManager() {
         return booleanFormulaManager;
     }
 
-    public Formula encodeFinalIntegerExpression(Expression expression) {
-        return new ExpressionEncoder(formulaManager, null).encodeAsInteger(expression);
+    public Formula encodeFinalExpression(Expression expression) {
+        return new ExpressionEncoder(this, null).encode(expression);
     }
 
     public BooleanFormula encodeBooleanExpressionAt(Expression expression, Event event) {
-        return new ExpressionEncoder(formulaManager, event).encodeAsBoolean(expression);
+        return new ExpressionEncoder(this, event).encodeAsBoolean(expression);
```

Comment:
Does this function convert integer expressions to boolean one's, or does it perform a type-check to make sure only boolean expressions get through?
If it is the former, I think the function should get renamed to something like `encodeExpressionAsBooleanAt`.
Otherwise it is fine.